extensions/blocks/google-calendar/edit.js

```diff
@@ -0,0 +1,193 @@
+/**
+ * External dependencies
+ */
+import { __, _x } from '@wordpress/i18n';
+import { Component } from '@wordpress/element';
+import {
+	Placeholder,
+	SandBox,
+	Button,
+	Notice,
+	ExternalLink,
+	PanelBody,
+} from '@wordpress/components';
+import { BlockIcon, InspectorControls } from '@wordpress/block-editor';
+
+/**
+ * Internal dependencies
+ */
+import { icon } from '.';
+import { isMobile } from '../../../_inc/client/lib/viewport';
+import { extractAttributesFromIframe, IFRAME_REGEX, URL_REGEX } from './utils';
+
+class GoogleCalendarEdit extends Component {
+	constructor() {
+		super( ...arguments );
+
+		this.state = {
+			editedEmbed: this.props.attributes.url || '',
+			editingUrl: false,
+			interactive: false,
+			notice: null,
+		};
+	}
+
+	static getDerivedStateFromProps( nextProps, state ) {
+		if ( ! nextProps.isSelected && state.interactive ) {
+			// We only want to change this when the block is not selected, because changing it when
+			// the block becomes selected makes the overlap disappear too early. Hiding the overlay
+			// happens on mouseup when the overlay is clicked.
+			return { interactive: false };
+		}
+
+		return null;
+	}
+
+	hideOverlay = () => {
+		// This is called onMouseUp on the overlay. We can't respond to the `isSelected` prop
+		// changing, because that happens on mouse down, and the overlay immediately disappears,
+		// and the mouse event can end up in the preview content. We can't use onClick on
+		// the overlay to hide it either, because then the editor misses the mouseup event, and
+		// thinks we're multi-selecting blocks.
+		this.setState( { interactive: true } );
+	};
+
+	setAttributes = event => {
```

Comment:
Could this lead to confusion with the `setAttributes` prop? Could this be something more descriptive like `handleEmbedEvent`?

Reply:
true, have renamed.